src/main/java/be/lghs/accounting/configuration/SecurityUserService.java

```diff
@@ -11,28 +13,49 @@
 
 import java.util.Map;
 import java.util.UUID;
+import java.util.stream.Collectors;
 
 @Service
 @RequiredArgsConstructor
 public class SecurityUserService extends DefaultOAuth2UserService {
 
     private final UserRepository userRepository;
 
+    private static final Logger logger = LoggerFactory.getLogger(SecurityUserService.class);
+
+    String getMandatory(Map<String, Object> values, String attribute){
+        if(values.containsKey(attribute)){
+            return (String) values.get(attribute);
+        } else {
+            logger.error("Missing attribute " + attribute + " for user " +
+                    values.entrySet().stream().map(e-> e.getKey() +":"+e.getValue())
+                            .collect(Collectors.joining(";")));
+
+            throw new RuntimeException("Missing attribute " + attribute + " for user ");
+        }
+    }
+
     @Override
     public OAuth2User loadUser(OAuth2UserRequest userRequest) throws OAuth2AuthenticationException {
-        OAuth2User user = super.loadUser(userRequest);
-        Map<String, Object> attributes = user.getAttributes();
-
-        UsersRecord usersRecord = userRepository.ensureUserExists(
-            (int) attributes.get("id"),
-            UUID.fromString((String) attributes.get("uuid")),
-            (String) attributes.get("name"),
-            (String) attributes.get("username"),
-            (String) attributes.get("email"));
-
-        return new OAuth2UserImpl(
-            usersRecord.getRoles(),
-            attributes
-        );
+        try {
+            OAuth2User user = super.loadUser(userRequest);
+            Map<String, Object> attributes = user.getAttributes();
+
+            UsersRecord usersRecord = userRepository.ensureUserExists(
+                    UUID.fromString(getMandatory(attributes, "sub")),
+                    getMandatory(attributes,"name"),
+                    getMandatory(attributes,"preferred_username"),
+                    getMandatory(attributes, "email"));
+
+            return new OAuth2UserImpl(
+                    usersRecord.getRoles(),
+                    attributes,
+                    usersRecord
+            );
+
+        } catch( Exception e){
```

Comment:
don't try/catch the exception, you already logged everything required in `getMandatory`

Reply:
there is other potential errors, I removed the log from getMandatory to not have a duplication of logs.

Reply:
There is indeed other potential error. Spring will handle all of them just fine, leave exception handling to the framework, it will log what's required and display to the user what's required.